File: 351003/Razhnou/task361/publisher/src/main/java/by/bsuir/task361/publisher/controller/UserV2Controller.java
package by.bsuir.task361.publisher.controller;

import by.bsuir.task361.publisher.dto.request.UserRequestTo;
import by.bsuir.task361.publisher.dto.response.UserResponseTo;
import by.bsuir.task361.publisher.service.SecuredUserService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/v2.0/users")
public class UserV2Controller {
    private final SecuredUserService securedUserService;

    public UserV2Controller(SecuredUserService securedUserService) {
        this.securedUserService = securedUserService;
    }

    @PostMapping
    public ResponseEntity<UserResponseTo> register(@Valid @RequestBody UserRequestTo request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(securedUserService.register(request));
    }

    @GetMapping
    public List<UserResponseTo> findAll() {
        return securedUserService.findAll();
    }

    @GetMapping("/{id}")
    public UserResponseTo findById(@PathVariable Long id) {
        return securedUserService.findById(id);
    }

    @PutMapping
    public UserResponseTo update(@Valid @RequestBody UserRequestTo request) {
        return securedUserService.update(request);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        securedUserService.delete(id);
        return ResponseEntity.noContent().build();
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/controller/UserV2Controller.kt
package com.distcomp.controller

import com.distcomp.dto.user.UserRequestTo
import com.distcomp.dto.user.UserResponseTo
import com.distcomp.service.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v2.0/users")
class UserV2Controller(
    private val userService: UserService
) {
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createUser(@Valid @RequestBody userRequestTo: UserRequestTo): UserResponseTo =
        userService.createUser(userRequestTo)

    @GetMapping("/{id}")
    fun readUserById(@PathVariable id: Long): UserResponseTo =
        userService.readUserById(id)

    @GetMapping
    fun findAll(): List<UserResponseTo> =
        userService.readAll()

    @PutMapping
    fun updateUser(@Valid @RequestBody userRequestTo: UserRequestTo): UserResponseTo =
        userService.updateUser(userRequestTo)

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteUser(@PathVariable id: Long) {
        userService.removeUserById(id)
    }
}